models/organization/org_times.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package organization
+
+import (
+	"code.gitea.io/gitea/models/db"
+	"xorm.io/builder"
+)
+
+// ResultTimesByRepos is a struct for DB query results
+type ResultTimesByRepos struct {
+	Name    string
+	SumTime int64
+}
+
+// ResultTimesByMilestones is a struct for DB query results
+type ResultTimesByMilestones struct {
+	RepoName     string
+	Name         string
+	ID           string
+	SumTime      int64
+	HideRepoName bool
+}
+
+// ResultTimesByMembers is a struct for DB query results
+type ResultTimesByMembers struct {
+	Name    string
+	SumTime int64
+}
+
+// GetTimesByRepos fetches data from DB to serve TimesByRepos.
+func (org *Organization) GetTimesByRepos(unixfrom, unixto int64) (results []ResultTimesByRepos, err error) {
```

Comment:
They don't need to become methods of Organization and they should get `context.Context` as the first parameter.

Reply:
Hmm, if the functions would not be a method of Organization, then it would have to take an `org_id` as a parameter, since it calculates on an Origanization basis (not globally). Would that be preferred, or should it stay an object function?

Reply:
The Organization object could be the first parameter of the function.  Thinking the difference between `func (org *Organization) GetTimesByRepos` and `func GetTimesByRepos(org *Organization`, the previous should be used when they are almost handling a property, otherwise use later. Of course, many places still didn't follow that which we need to do refactor.